neard/src/main.rs

```diff
@@ -42,7 +42,33 @@ static ALLOC: MyAllocator<tikv_jemallocator::Jemalloc> =
 #[global_allocator]
 static ALLOC: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
 
+fn prog() -> Option<String> {
+    env::args()
+        .next()
+        .as_ref()
+        .map(std::path::Path::new)
+        .and_then(std::path::Path::file_name)
+        .and_then(std::ffi::OsStr::to_str)
+        .map(String::from)
+}
+
 fn main() {
+    if prog().map(|name| name == "near").unwrap_or_default() {
+        println!("WARNING WARNING WARNING");
+        println!("WARNING WARNING WARNING");
+        println!("WARNING WARNING WARNING");
+        println!("WARNING WARNING WARNING");
+        println!("WARNING WARNING WARNING");
+        println!("WARNING WARNING WARNING");
+        println!("WARNING WARNING WARNING");
+        println!("WARNING WARNING WARNING");
+        println!("WARNING WARNING WARNING");
+        println!("Usage of `./near` binary is deprecated since May 2020!!!");
+        println!("Use `./neard` instead");
+        println!("Use ./neard build target");
```

Comment:
`tracing::error!(…)` or `tracing::warn!(…)` would be better than
`println!`.  Also, the message should better describe what is going
on.  Perhaps something like:

```
use tracing::warn;
```
```
        warn!(target: "neard",
              "This executable has been compiled as a ‘near’ bin target.");
        warn!(target: "neard",
              "That target has been deprecated since May 2020 and will \
               eventually be removed.");
        warn!(target: "neard",
              "Please update your build process to use the new ‘neard’ target,\
               e.g. by using ‘make release’ or ‘make neard’ commands.");
        warn!(target: "neard", "See {} for more information.",
              "https://docs.near.org/docs/develop/node/validator/compile-and-run-a-node");
        warn!(target: "neard",
              "I’m now going to sleep for 30 seconds to be obnoxious about \
               it.  ;)");
```